datatypes/src/operations/image/colorizer.rs

```diff
@@ -517,7 +517,7 @@ pub type Breakpoints = Vec<Breakpoint>;
 /// It is assumed that is has at least one and at most 256 entries.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(try_from = "SerializablePalette", into = "SerializablePalette")]
-pub struct Palette(HashMap<NotNan<f64>, RgbaColor>);
+pub struct Palette(pub HashMap<NotNan<f64>, RgbaColor>);
```

Comment:
Why do we need to make this pub?

Reply:
the Palette is part of our API but defined in the datatypes crate. Thus I copied it to the services crate and implemented a From-Conversion. But that requires access to the map.

Reply:
if you have an idea, please let me know. I couldn't think of one.

Reply:
At least create a special constructor for that. Then, you don't access the fields differently now.

Reply:
`pub` means you can access the internal HashMap, etc. directly.
That wasn't the case before.

Reply:
so do you want an *accessor*  or a constructor?

Reply:
You can remove your pub declaration of the internals of the struct. This means, you can only work with the struct like before and not access the internals directly.